### PR TITLE
docs(visualRegression): Basic how to

### PR DESCRIPTION
**Pull request by Droogans**

### CODE REVIEW

File: guides/testing.md [hunk 1]
```diff
@@ -185,3 +185,32 @@ Click the reference link to the encore-ui-screenshots pull request, and check ou
 ### Creating a New Baseline for Visual Regression
 
 Once a pull request has been visually signed off on, it should become the new baseline which all new pull requests are compared against. Do this by visiting the link to the encore-ui-screenshots pull request and merging it. From then on, Travis CI pull request builds will pull *those* screenshots and compare against them from now on.
+
+### Creating New Visual Regression Tests
+
+The visual regression tests are located in [the visual-regression directory](../utils/visual-regression), which is in `utils`. Inside there is a directory for `layouts` (typically reserved for pages that demonstrate how a typical EncoreUI layout should look), and a directory for components.
```
Comment (CITguy) [inline guides/testing.md]: ```..., which is in `utils`.``` seems repetitive. 

Maybe try, 
```
The visual regression tests are located in the [`/utils/visual-regression`](../utils/visual-regression) directory.
```

File: guides/testing.md [hunk 2]
```diff
@@ -185,3 +185,32 @@ Click the reference link to the encore-ui-screenshots pull request, and check ou
...
+
+When adding a new visual regression test, remember to keep the names of the tests short. The screenshot comparison library creates the image names based on the test names, so there's no need to be very specific.
+
+> *utils/visual-regression/components/rxComponent.midway.js*
```
Comment (CITguy) [inline guides/testing.md]: I'd put a leading forward slash (`/utils/...`)

Comment (CITguy) [inline guides/testing.md]: I'd put this line as a comment at the top of the code block below.

**Comment by parlarjb** [comment]
I feel like the code needs more of an introduction/explanation.

Something like:

Each `it()` block describes an individual visual regression test. `screenshot.snap()` is used to grab an area of the screen, and compare it to the current visual baseline, checking for changes. Any detected change will cause a visual diff to be created, with a link to it automatically added to the PR.

File: guides/testing.md (hunk 1 above)
Comment (Droogans) [inline guides/testing.md]: Yes, it's awkward. Good suggestion.

File: guides/testing.md (hunk 2 above)
Comment (Droogans) [inline guides/testing.md]: Sure.

**Comment by Droogans** [comment]
Ok I'll summarize that and include it as well.

**Comment by parlarjb** [comment]
@Droogans ping ping

**Comment by Droogans** [comment]
@parlarjb It's here now.

**Comment by parlarjb** [comment]
Looks good. Please squash, and we'll merge it in.

**Comment by Droogans** [comment]
@parlarjb squashed.

File: guides/testing.md [hunk 3]
```diff
@@ -38,7 +38,11 @@ If you wish to run them separately, use `grunt test`.
 
 When developing a specific component, you likely don't want to run the entire test suite on every change. In order to test a single set of functionality, use the 'only' function when describing your test. For example:
 
-`describe.only('Login', function () { ... tests go here ... })`
+```js
+describe.only('Login', function () {
+    // ... tests go here ...
+})
```
Comment (CITguy) [inline guides/testing.md]: semicolon missing

File: guides/testing.md [hunk 4]
```diff
@@ -185,3 +189,37 @@ Click the reference link to the encore-ui-screenshots pull request, and check ou
...
+
+When adding a new visual regression test, remember to keep the names of the tests short. The screenshot comparison library creates the image names based on the test names, so there's no need to be very specific.
+
+Each `it()` block describes an individual visual regression test. `screenshot.snap()` is used to grab an area of the screen, and compare it to the current visual baseline, checking for changes. Any detected change will cause a visual diff to be created, with a link to it automatically added to the PR.
+
+Here's an example of a basic visual regression test for a single component.
+
+```js
+// utils/visual-regression/components/rxComponent.midway.js
+var rxComponent = require('../../../src/tabs/rxComponent.page').rxComponent;
```
Comment (CITguy) [inline guides/testing.md]: Does this work with the new syntax added with the recent updates to page objects?

Comment (Droogans) [inline guides/testing.md]: Ah! Thanks. Good catch.

File: guides/testing.md (hunk 3 above)
Comment (Droogans) [inline guides/testing.md]: Is it syntactically *required*? </snark>

I'll add it. Good catch.

**Comment by Droogans** [comment]
@parlarjb @CITguy I went ahead and amended + force pushed to avoid the rebase.

**Comment by parlarjb** [comment]
LGTM

**Comment by megalord** [comment]
LGTM
The only question this doesn't answer for me is where `screenshot` comes from.  I'm assuming it's global?

**Comment by Droogans** [comment]
It's in the `onPrepare` hook in the [protractor visual regression conf file](https://github.com/rackerlabs/encore-ui/blob/4a4ba7053717ebbfe2e7a26b4823e19e9a860e6d/protractor.visual.regression.conf.js#L28).

**Pull request merged**